_posts/2016-10-25-what-is-devsec.md

```diff
@@ -0,0 +1,114 @@
+* Operating system security
+* Application server security
+* Language and framework specific security 
+* Component security
+* Code
+* Data integrity
+
+Sofware architects and developers design software architecture that affects these issues. Security should be one consideration when operating systems and technology are chosen for a software system. The development team usually doesn't need to worry about BIOS security or physical security of the data center, but many security controls are their responsibility.
+
+## What is DevSec?
+
+Like DevOps, this is about the culture. It's about responsibility. It's about knowing the right thing and
+having the power, resources and ability to do it. And of course It's about hacking yourself first. A somewhat mechanical implementation would be to add a
+"security specialist" to development team, as you could add some "system operation specialist" and say you're now
+doing DevOps. In our opinion, this approach is a minor improvement and totally misses the point about *culture*. A culture is about 
+values and beliefs shared by everyone. It is not enough that a "security expert" sits with the team, if that person
+is the only one concerned about security. The DevSec goal is to make every developer think about security and acknowledge 
+it is necessary. Not everyone needs to be an expert, but everyone needs to understand that security is essential and important.
+
+## Security conscious developers
+
+As we said, DevSec calls for developers to become conscious about security. We recommend a secure development lifecycle model as a starting point for practical everyday work. It dictates what needs to be dealt with, but the team decides what is the proper thing to do in their particular context. If you are a wildling, and do not like processes, you can also try "hack yourself first" motto as your starting point. Here's an example list of actions and decisions to make:
+
+* Train your personnel.
+* Know the policies and laws that affect your project.
+* Document your architecture and understand your attack surface.
+* Do a threat analysis and risk assesment.
+* Follow vulnerability feeds for your technology stack.
+* Keep on track about dependencies and their possible vulnerabilities.
+* Know the best practices with your technologies.
+* Analyze your code and configurations.
+* Do inhouse penetration testing, try to break your system.
+* Monitor your application.
+* Automate alerts for anomalies in the usage patterns.
+* Do incident & response handling.
+* Prepare and plan for compromise.
+* Revalidate on changes.
+
+It's important to understand that cybercriminals estimate payoff/risk and payoff/effort before they seriously attack a system. If your system is difficult to crack, it won't get attacked unless there's a very big payoff. The criminals take the easy money, always.  Hackers and hacktivists might have other motivations, but here's a rough model for calculating "criminal value" of a system:
```

Comment:
Opportunistic low-hanging-fruit pickers are the most common type of attack seen, but targeted attacks are motivated by risk vs return too - eg hired guns enlisted by competitors, intelligence actors, etc. They pick their targets, and in the target they pick the easiest way to get to the organisation.